.github/workflows/deprecate-releases.yaml

```diff
@@ -0,0 +1,216 @@
+name: Deprecate Outdated Releases
+
+on:
+  # Run weekly on Monday morning
+  schedule:
+    - cron: "0 6 * * 1" # Every Monday at 6:00 AM UTC
+
+  # Allow manual triggering
+  workflow_dispatch:
+  push:
+    branches:
+      - "master"
+      - "main"
+      - "release-deprecation"
+    paths:
+      - ".github/workflows/deprecate-releases.yaml"
+      - "azure/**"
+      - "capa/**"
+      - "cloud-director/**"
+      - "vsphere/**"
+
+jobs:
+  find-releases:
+    runs-on: ubuntu-latest
+    outputs:
+      providers: ${{ steps.find-providers.outputs.providers }}
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Find providers with releases
+        id: find-providers
+        run: |
+          PROVIDERS=$(find . -maxdepth 1 -type d -name "azure" -o -name "capa" -o -name "vsphere" -o -name "cloud-director" | sed 's|./||' | jq -R -s -c 'split("\n") | map(select(length > 0))')
+          echo "providers=$PROVIDERS" >> $GITHUB_OUTPUT
+
+  deprecate-releases:
+    needs: find-releases
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        provider: ${{ fromJson(needs.find-releases.outputs.providers) }}
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Set up Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: "1.24"
+
+      - name: Set provider display name
+        id: provider-name
+        run: |
+          PROVIDER="${{ matrix.provider }}"
+          case "$PROVIDER" in
+            capa)
+              echo "display_name=CAPA" >> $GITHUB_OUTPUT
+              ;;
+            azure)
+              echo "display_name=CAPZ" >> $GITHUB_OUTPUT
+              ;;
+            vsphere)
+              echo "display_name=CAPV" >> $GITHUB_OUTPUT
+              ;;
+            cloud-director)
+              echo "display_name=CAPVCD" >> $GITHUB_OUTPUT
+              ;;
+            *)
+              echo "display_name=${PROVIDER^^}" >> $GITHUB_OUTPUT
+              ;;
+          esac
+
+      - name: Build deprecation tool
+        run: |
+          cd tools
+          go mod tidy
+          go build -o deprecate-releases tools/deprecate-release.go
```

Comment:
Not sure `tools/` is needed here is it? We're already in the tools dir

Reply:
Yeah I noticed it :D I removed it

Reply:
You're too quick for me! 😆